elastic/synthetics-recorder
# fix: Fix bugs around tooltip for Test button

## Summary
Tooltips on `Test` button works weirdly after running test. The tooltip appears as soon as pressing the button with the wrong message and also it won't go away even after the flyout shows up:
[![Image from Gyazo](https://i.gyazo.com/93340e1dfcba6995cbdbd13013db92d3.gif)](https://gyazo.com/93340e1dfcba6995cbdbd13013db92d3)
Also after the test run, when resizing to a smaller width, the Test tooltip would be visible all the time rather than visible only when hovered:
[![Image from Gyazo](https://i.gyazo.com/ce562760cbd1c25855bf1c6309c7152f.gif)](https://gyazo.com/ce562760cbd1c25855bf1c6309c7152f)


This PR fixes the tooltip for the Test button to be visible only when it's supposed to be. For example, there are 3 conditions that make `Test` button disabled but now we display the same message applies to only one condition to all the cases. This fix prevents that and shows the tooltip with the correct description only when it's applicable.
[![Image from Gyazo](https://i.gyazo.com/426ff936c9a43f93a74533dcfe816308.gif)](https://gyazo.com/426ff936c9a43f93a74533dcfe816308)

Also, it fixes the behavior of the tooltip for Test button being always visible after running the test run:
[![Image from Gyazo](https://i.gyazo.com/bac76514742c0a1959b85ecbaa9eb964.gif)](https://gyazo.com/bac76514742c0a1959b85ecbaa9eb964)

## How to validate this change
Record a script, press the `Test` button, and confirm that the tooltip doesn't show up on Test button. After the test run is finished, try resizing until the button shows only the icons and confirm the Test tooltip is only visible when the mouse is over the button.
